float32 distance_info
